src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        size = 0;
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
+
+    private static class Node<K, V> {
+        K key;
+        V value;
+        Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> node;
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            node = table[index];
+            if (node.key == null || node.key.equals(key)) {
+                node.value = value;
+                return;
+            }
+            while (node.next != null) {
+                node = node.next;
+                if (node.key.equals(key)) {
+                    node.value = value;
+                    return;
+                }
+            }
+            node = new Node<>(key, value, table[index]);
+            table[index] = node;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return table[0].value;
+        }
+        for (Node<K, V> node : table) {
```

Comment:
method get() in map should have complexity O(1), if you use for each it will have O(n), cause your should iterate through whole array, use getIndex here instead